README.md

```diff
@@ -487,6 +487,142 @@ To begin coding the basic execution functionality of our VM, we only need these
+
+```go
+// Updates the value of PC according to the executed instruction
+func (vm *VirtualMachine) UpdatePc(instruction *Instruction, operands *Operands) error {
+	switch instruction.PcUpdate {
+	case PcUpdateRegular:
+		vm.RunContext.Pc.Offset += instruction.Size()
+	case PcUpdateJump:
+		if operands.Res == nil {
+			return errors.New("Res.UNCONSTRAINED cannot be used with PcUpdate.JUMP")
+		}
+		res, ok := operands.Res.GetRelocatable()
+		if !ok {
+			return errors.New("An integer value as Res cannot be used with PcUpdate.JUMP")
+		}
+		vm.RunContext.Pc = res
+	case PcUpdateJumpRel:
+		if operands.Res == nil {
+			return errors.New("Res.UNCONSTRAINED cannot be used with PcUpdate.JUMP_REL")
+		}
+		res, ok := operands.Res.GetFelt()
+		if !ok {
+			return errors.New("A relocatable value as Res cannot be used with PcUpdate.JUMP_REL")
+		}
+		new_pc, err := vm.RunContext.Pc.AddFelt(res)
+		if err != nil {
+			return err
+		}
+		vm.RunContext.Pc = new_pc
+    case PcUpdateJnz:
+		if operands.Dst.IsZero() {
+			vm.RunContext.Pc.Offset += instruction.Size()
+		} else {
+			new_pc, err := vm.RunContext.Pc.AddMaybeRelocatable(operands.Op1)
+			if err != nil {
+				return err
+			}
+			vm.RunContext.Pc = new_pc
+		}
+
+	}
+	return nil
+}
+```
+
+Some auxiliary methods where added for this method:
+
+*Instruction.Size*
+
+Returns 1 if the instruction has no immediate value or 2 if it has. We can tell that an instruction has an immediate value if the op1 address is given by the immediate value.
+
+```go
+func (i *Instruction) Size() uint {
+	if i.Op1Addr == Op1SrcImm {
+		return 2
+	}
+	return 1
+}
+```
+
+*MaybeRelocatable.IsZero()*
+
+Returns true if the value is a Felt that is zero, returns false otherwise
+
+```go
+    func (m *MaybeRelocatable) IsZero() bool {
+	felt, is_int := m.GetFelt()
+	return is_int && felt.IsZero()
+}
+```
+
+*UpdateFp*
+
+As we already know, the fp (frame pointer) points to the frame of the current function. It can be updated in 4 different ways. A regular fp update means changes to the fp register. An ap plus 2 update consists on asigning the value op ap to fp and increasing it's offset by two (note: in the code below we only asign the offset, as fp and ap live on the execution segment and therefore have the same segment index). A dst fp update consists on performing either a direct or relative jump based on the value of the dst operand. If dst is a relocatable, fp will take the value of dst, if dst is a felt, fp's offset will be increased by the amount given by dst
```

Comment:
```suggestion
As we already know, the fp (frame pointer) points to the frame of the current function. It can be updated in 4 different ways. A regular fp update means no changes to the fp register. An ap plus 2 update consists on asigning the value of ap to fp and increasing it's offset by two (note: in the code below we only assign the offset, as fp and ap live on the execution segment and therefore have the same segment index). A dst fp update consists in performing either a direct or relative jump based on the value of the dst operand. If dst is a relocatable, fp will take the value of dst, if dst is a felt, fp's offset will be increased by the amount given by dst
```